src/app/features/courses/edit-page/edit-page.component.ts

```diff
@@ -0,0 +1,110 @@
+import {Component, OnInit} from '@angular/core';
+import {FormArray, FormBuilder, FormControl, FormGroup, Validators} from "@angular/forms";
+import {ActivatedRoute, Router} from "@angular/router";
+import {CoursesStoreService} from "../../../services/courses-store.service";
+import {Course} from "../../../services/course";
+import {CoursesService} from "../../../services/courses.service";
+
+@Component({
+  selector: 'app-edit-page',
+  templateUrl: './edit-page.component.html',
+  styleUrls: ['./edit-page.component.css']
+})
+export class EditPageComponent implements OnInit {
+  isEditMode: boolean = false;
+  courseId: string | null = '';
+  public formData = {
+    title: '',
+    description: '',
+    authorName: '',
+    authors: [],
+    duration: 0,
+  }
+
+  editForm: FormGroup;
+
+  constructor(private fb: FormBuilder, private route: ActivatedRoute, private coursesService: CoursesService, private router: Router) {
+    this.editForm = new FormGroup({
+      title: new FormControl(this.formData.title, [
+        Validators.required,
+        Validators.minLength(4)
+      ]),
+      description: new FormControl(this.formData.description, [
+        Validators.required,
+        Validators.minLength(10)
+      ]),
+      authorName: new FormControl(this.formData.authorName, [
+        Validators.required,
+        Validators.minLength(4)
+      ]),
+      authors: new FormArray([]),
+      duration: new FormControl(this.formData.duration, [
+        Validators.required,
+        Validators.min(0)
+      ]),
+    })
+  }
+
+  ngOnInit(): void {
+    this.courseId = this.route.snapshot.paramMap.get('id');
+
+    if (this.courseId) {
+      this.isEditMode = true;
+      this.coursesService.getCourse(this.courseId).subscribe(data => {
+        this.title?.setValue(data.title);
+        this.description?.setValue(data.description);
+        this.duration?.setValue(data.duration);
+        this.authors?.setValue(data.authors);
+      })
+    }
+  }
+
+  get title() {
+    return this.editForm.get('title')!;
+  }
+
+  get description() {
+    return this.editForm.get('description')!;
+  }
+
+  get duration() {
+    return this.editForm.get('duration')!;
+  }
+
+  get authorName() {
+    return this.editForm.get('authorName')!;
+  }
+
+  get authors() {
+    return this.editForm.controls['authors'] as FormArray;
+  }
+
+  addAuthor() {
+    const authorForm = this.fb.group({
+      name: [this.editForm.get('authorName')?.value]
+    });
+    this.authors.push(authorForm);
+  }
+
+  deleteAuthor(authorIndex: number) {
+    this.authors.removeAt(authorIndex);
+  }
+
+  updateCourse() {
+    this.coursesService.editCourse(this.editForm.value.title, this.editForm.value.description,
```

Comment:
Better top use course model here. According to Clean code rules we should avoid method with more than 3 args.